InnerEye/ML/Histopathology/datasets/panda_tiles_dataset.py

```diff
@@ -54,17 +54,26 @@ def __init__(self,
                  transform: Optional[Callable] = None,
                  **kwargs: Any) -> None:
         super().__init__(root=root, transform=transform)
-        self.base_dataset = PandaTilesDataset(root=root,
+        # filter out tiles based on occupancy threshold
+        self.full_dataset = PandaTilesDataset(root=root,
                                               dataset_csv=dataset_csv,
                                               dataset_df=dataset_df)
+        self.occupancy_threshold = 0
+        dataset_df_filtered = self.full_dataset.dataset_df.loc[self.full_dataset.dataset_df['occupancy'] > self.occupancy_threshold]  # type: ignore
```

Comment:
I'd suggest to move this filtering to the `PandaTilesDataset.__init__()` with `occupancy_threshold` as an optional argument (e.g. no filtering if `None`). This way, you can simply do `self.base_dataset = PandaTilesDataset(..., occupancy_threshold=xxx)` here for SSL, but it also enables this filtering for other uses of the dataset like MIL.

Reply:
Thanks, Daniel. Could you please explain a bit more how to include filtering in the constructor, as first I instantiate the tiles dataset to get the full dataset_df which is then filtered based on occupancy. The dataset_df is not yet assigned in the constructor.

Reply:
You can override `self.dataset_df` after `super().__init__(...)`. This way, the full dataframe is already created and you can filter it.

Reply:
I have now made this change. Thanks for the nice idea!